src/Fixer/Comment/HeaderCommentFixer.php

```diff
@@ -19,58 +19,74 @@
 
 /**
  * @author Antonio J. García Lagar <aj@garcialagar.es>
+ * @author SpacePossum
  */
 final class HeaderCommentFixer extends AbstractFixer
 {
-    private $header = '';
-    private $headerComment = '';
+    const HEADER_PHPDOC = 1;
+    const HEADER_COMMENT = 2;
 
-    /**
-     * {@inheritdoc}
-     */
-    public function configure(array $configuration = null)
-    {
-        if (null === $configuration || !isset($configuration['header'])) {
-            throw new InvalidFixerConfigurationException($this->getName(), 'Configuration is required.');
-        }
+    const HEADER_LOCATION_AFTER_OPEN = 1;
+    const HEADER_LOCATION_AFTER_DECLARE_STRICT = 2;
 
-        $header = $configuration['header'];
+    const HEADER_LINE_SEPARATION_BOTH = 1;
+    const HEADER_LINE_SEPARATION_TOP = 2;
+    const HEADER_LINE_SEPARATION_BOTTOM = 3;
+    const HEADER_LINE_SEPARATION_NONE = 4;
 
-        if (!is_string($header)) {
-            throw new InvalidFixerConfigurationException($this->getName(), sprintf('Header configuration is invalid. Expected "string", got "%s".', is_object($header) ? get_class($header) : gettype($header)));
-        }
+    /** @var string */
+    private $headerComment;
 
-        $this->setHeader($header);
-    }
+    /** @var int */
+    private $headerCommentType;
+
+    /** @var int */
+    private $headerLocation;
+
+    /** @var int */
+    private $headerLineSeparation;
 
     /**
      * {@inheritdoc}
      */
-    public function isCandidate(Tokens $tokens)
+    public function configure(array $configuration = null)
```

Comment:
Please, describe the configuration array. For now I have no idea what to pass here